src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -168,59 +227,12 @@ final class CachedAppConfiguration {
 		for promise in self.promises {
 			promise(result)
 		}
-		self.promises = [(Result<CachedAppConfiguration.AppConfigResponse, Never>) -> Void]()
-	}
-}
-
-extension CachedAppConfiguration: AppConfigurationProviding {
-
-	fileprivate static let timestampKey = "LastAppConfigFetch"
-
-	func appConfiguration(forceFetch: Bool = false) -> AnyPublisher<SAP_Internal_V2_ApplicationConfigurationIOS, Never> {
-		let force = shouldFetch() || forceFetch
 
-		if let cachedVersion = store.appConfigMetadata?.appConfig, !force {
-			Log.debug("fetching cached app configuration", log: .appConfig)
-			// use the cached version
-			return Just(cachedVersion)
-				.receive(on: DispatchQueue.main.ocombine)
-				.eraseToAnyPublisher()
-		} else {
-			Log.debug("fetching fresh app configuration. forceFetch: \(forceFetch), force: \(force)", log: .appConfig)
-			// fetch a new one
-			return getAppConfig(with: store.appConfigMetadata?.lastAppConfigETag)
-				.receive(on: DispatchQueue.main.ocombine)
-				.map({ $0.config })
-				.eraseToAnyPublisher()
+		if case let .success(appConfigResponse) = result,
```

Comment:
Would be nice to have this in an own func to separate logic. Or maybe rename the func.